src/docfx/restore/PackageResolver.cs

```diff
@@ -174,6 +177,70 @@ private void DownloadGitRepositoryCore(string cwd, string url, string committish
             }
         }
 
+        private void ThrowRestoreDependentRepositoryFailureError(string url, string committish, Exception ex)
+        {
+            var systemServiceTokenAccountType = "SystemServiceAccount";
+            var templateRepoUrlPrefix = "https://github.com/Microsoft/templates.docs.msft";
+
+            if (ex.Message.Contains("has enabled or enforced SAML SSO", StringComparison.OrdinalIgnoreCase))
+            {
+                if (url.StartsWith(templateRepoUrlPrefix) || _config.GitAccessTokenAccountType.Equals(systemServiceTokenAccountType))
+                {
+                    throw Errors.System.GitCloneFailed(url, committish).ToException(ex);
+                }
+                else
+                {
+                    throw Errors.CRR.RepositoryOwnerSSOIssue(EnvironmentVariable.RepositoryUrl ?? "", _config.RepositoryOwnerName, url).ToException(ex);
```

Comment:
The recommended way to retrieve current repository URL is to access the `Repository.Remote` property. We don't always set `DOCFX_REPOSITORY_URL`. You can constructor inject an instance of `Repository`.


```suggestion
                    throw Errors.CRR.RepositoryOwnerSSOIssue(_repository.Remote, _config.RepositoryOwnerName, url).ToException(ex);
```